vmm/src/device_manager.rs

```diff
@@ -348,6 +348,9 @@ pub enum DeviceManagerError {
     /// Failed to remove a bus device from the MMIO bus.
     RemoveDeviceFromMmioBus(vm_device::BusError),
 
+    /// Couldn't find the barrier associated with the PCI device being removed.
+    RemoveDeviceMissingBarrier,
```

Comment:
@sboeuf, correct me if I'm mistaken, but this is a **workaround** for the missing barrier, right?
If so, where should be the right place to add this one?

It's a good workaround and we can update this once the barrier is added to the correct place, but I'm curious. :-)

Reply:
Oh no this is not a workaround, this is the error type that must be returned in case the `Barrier` can't be found in the `BTreeMap` with the corresponding `PciBdf` key.